dev-libs/libbpf-0.3 7  dev-libs/libbpf-9999  (required by dev-util/bpftrace-9999) build fails:

x86_64-pc-linux-gnu-gcc -shared -Wl,--version-script=libbpf.map \
	  -Wl,-soname,libbpf.so.0 \
	  sharedobjs/bpf.o sharedobjs/btf.o sharedobjs/libbpf.o sharedobjs/libbpf_errno.o sharedobjs/netlink.o sharedobjs/nlattr.o sharedobjs/str_error.o sharedobjs/libbpf_probes.o sharedobjs/bpf_prog_linfo.o sharedobjs/xsk.o sharedobjs/btf_dump.o sharedobjs/hashmap.o sharedobjs/ringbuf.o -Wl,-O1 -Wl,--as-needed -march=native -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=6 -fuse-linker-plugin -fstack-clash-protection -fcf-protection=full -pipe -lelf -lz  -o libbpf.so.0.3.0
{standard input}: Assembler messages:
{standard input}: Error: invalid attempt to declare external version name as default in symbol `xsk_umem__create@@LIBBPF_0.0.4'
make[1]: *** [/var/tmp/portage/dev-libs/libbpf-0.3/temp/cc7USKSk.mk:2: /var/tmp/portage/dev-libs/libbpf-0.3/temp/libbpf.so.0.3.0.U9ULWi.ltrans0.ltrans.o] Error 1
make[1]: *** Waiting for unfinished jobs....
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make: *** [Makefile:97: libbpf.so.0.3.0] Error 1
 * ERROR: dev-libs/libbpf-0.3::gentoo failed (compile phase):
 *   emake failed
 * 

after switching to 
CFLAGS="-march=native  -O2 -pipe"
builds correctly

libbpf-9999 build.log:
https://pastebin.com/raw/GBemeMtt
libbpf-0.3 buillog:
https://pastebin.com/raw/C2HcavUX

Cheers

--- a/sys-config/ltoize/files/package.cflags/lto.conf
+++ b/sys-config/ltoize/files/package.cflags/lto.conf
@@ -31,6 +31,7 @@ dev-lang/rust *FLAGS-=-flto*
 dev-lang/spidermonkey *FLAGS-=-flto*
 dev-libs/elfutils *FLAGS-=-flto*
 dev-libs/intel-neo *FLAGS-=-flto* # error: violates the C++ One Definition Rule [-Werror=odr]
+dev-libs/libbpf *FLAGS-=-flto* # lto-wrapper link failure
 dev-libs/libsigsegv *FLAGS-=-flto* # Issue #189
 dev-libs/weston *FLAGS-=-flto*
 dev-python/libvirt-python *FLAGS-=-flto* # ICE with GCC 10.2.0

